Documentation fixes

pindec opened the pull request:
- Fixes #131 - updating mapping page 
- Fixes #146 - add 'publicAuthority' role to example file
- Fixes #136 - add project identifier prefix to example file
- Fixes #143 - adding project package, metadata to worked example pages; link to OCDS guidance plus improvements to example page

pindec committed:
fix for #131 replace owner with publicAuthority

pindec committed:
fix for #146 add publicAuthority role to worked example

pindec committed:
fix for #136 add oc4ids example prefix to worked example

pindec committed:
fix for #143 adding package, metadata to worked example and improving example text

duncandewhurst commented on docs/guidance/example.md:
```diff
@@ -4,32 +4,63 @@ The [OC4IDS schema](../reference/index.md) sets out the structure and format of
-* **Documents** - documents and documentation relating to this project.
-* **Contracting processes** - information about related contracting processes for different aspects of the project.
-* **Completion** - information on the final scope, duration and costs for the project.
+* **Package metadata** - a container for data on multiple projects with important metadata about the publication
```
Package metadata > Project package

duncandewhurst commented on docs/guidance/example.md:
```diff
@@ -4,32 +4,63 @@ The [OC4IDS schema](../reference/index.md) sets out the structure and format of
-* **Contracting processes** - information about related contracting processes for different aspects of the project.
-* **Completion** - information on the final scope, duration and costs for the project.
+* **Package metadata** - a container for data on multiple projects with important metadata about the publication
+* **Project information** - for each project including:
```
Project information > Project data

duncandewhurst commented on docs/guidance/example.md:
```diff
@@ -4,32 +4,63 @@ The [OC4IDS schema](../reference/index.md) sets out the structure and format of
+## Sections
 
-The full JSON file for the example project looks like this:
+### Package metadata
```
Package metadata > Project package

duncandewhurst commented on docs/guidance/example.md:
```diff
@@ -4,32 +4,63 @@ The [OC4IDS schema](../reference/index.md) sets out the structure and format of
 
-The full JSON file for the example project looks like this:
+### Package metadata
+This section provides metadata about the data, including the schema version, publisher, data license and publication policy, and acts as a container for data on multiple projects in a `projects` array.
```
This section > The project package

duncandewhurst commented on docs/guidance/example.md:
````diff
@@ -4,32 +4,63 @@ The [OC4IDS schema](../reference/index.md) sets out the structure and format of
+```
+#### License
+
+The `license` field should contain a link to an appropriate data license. Further information about licensing can be found in our [OCDS guidance](https://standard.open-contracting.org/latest/en/implementation/licensing/).
````
an appropriate data license > the license that applies to the data in the package.

our [OCDS guidance] > the [OCDS licensing guidance]

duncandewhurst commented on docs/guidance/example.md:
```diff
@@ -4,32 +4,63 @@ The [OC4IDS schema](../reference/index.md) sets out the structure and format of
+
+#### Publication policy
+
+The `publicationPolicy` field should contain a link to an OC4IDS publication policy with similar information to an [OCDS publication policy](https://standard.open-contracting.org/latest/en/implementation/publication_policy/). Please refer to that documentation for more details.
```
Reword as:

> The `publicationPolicy` field should contain a link to guide for data users. For more information about what to include in a publication policy, refer to the [OCDS publication policy guidance](https://standard.open-contracting.org/latest/en/implementation/publication_policy/) and the [OCDS publication policy template](https://www.open-contracting.org/resources/ocds-1-1-publication-policy-template/).

duncandewhurst commented on docs/guidance/example.md:
```diff
@@ -4,32 +4,63 @@ The [OC4IDS schema](../reference/index.md) sets out the structure and format of
+### Project information
 
-### Project metadata
+An OC4IDS document may contain information about multiple infrastructure projects. Each infrastructure project is represented as an entry in the `projects` array. The example contains information about one infrastructure project.
```
An OC4IDS document > A project package

duncandewhurst commented on docs/guidance/example.md:
```diff
@@ -4,32 +4,63 @@ The [OC4IDS schema](../reference/index.md) sets out the structure and format of
+#### Project metadata
+
+This section provides contextual information about the infrastructure project, including:
 
```
Add:

> * A project identifier in the `id` field. To make the project identifier globally unique, a project identifier prefix must be added to the local identifier for the project. Project identifier prefixes are assigned by the OC4IDS helpdesk. For more information on project identifiers, refer to the [project identifiers guidance](../identifiers.md)

duncandewhurst commented on docs/guidance/example.md:
```diff
@@ -44,21 +75,21 @@ This section provides contextual information about the project, including:
    :jsonpointer: /projects/0/locations
    :expand:  geometry, gazetteer, address
-   :title: location 1
+   :title: location1
```
Is there a reason not just to use 'location'?

I suggest updating the previous paragraph to read:

> In this example, one location is given: a motorway junction, using a point location, a gazetteer entry and an address.

duncandewhurst commented on docs/guidance/example.md:
```diff
@@ -44,21 +75,21 @@ This section provides contextual information about the project, including:
-### Budget and budget breakdown
+#### Budget and budget breakdown
 
 The `budgetBreakdown` array in the `budget` section can be used to provide a breakdown of the budget by period and/or funding source.
```
Reword as:

> The `budget` section can be used to provide the overall budget for the project and a breakdown of the budget by period and/or funding source, in the `budgetBreakdown` array.

duncandewhurst commented on docs/guidance/example.md:
```diff
@@ -44,21 +75,21 @@ This section provides contextual information about the project, including:
 The `budgetBreakdown` array in the `budget` section can be used to provide a breakdown of the budget by period and/or funding source.
 
-In this example, the overall budget for the project covers 3 years and is broken down into amounts per year, with £10,000,000 allocated in 2016, as shown in the first `budgetBreakdown` component.
+In the budget example shown below, the overall budget for the infrastructure project covers 3 years and is broken down into amounts per year, with £10,000,000 allocated in 2016, as shown in the budgetBreakdown_1 example.
```
budgetBreakdown_1 example > budgetBreakdown example

duncandewhurst commented on docs/guidance/example.md:
```diff
@@ -68,70 +99,71 @@ In this example, the overall budget for the project covers 3 years and is broken
      :jsonpointer: /projects/0/budget/budgetBreakdown/0
      :expand: amount, period, sourceParty
-     :title: budgetBreakdown
+     :title: budgetBreakdown_1
```
budgetBreakdown_1 > budgetBreakdown

duncandewhurst commented on docs/guidance/example.md:
```diff
@@ -68,70 +99,71 @@ In this example, the overall budget for the project covers 3 years and is broken
+The `parties` array is used to provide details about organizations involved in the infrastructure project and their roles. Organization references elsewhere in the data refer back to entries in this section.
 
-In this example, details are given about the fictional Motorways UK entity, which is referred to from the `sourceParty` section in the `budgetBreakdown` example above, using the `name` and `id` from the entry in the `parties` section.
+In the party_1 example below, details are given about the fictional Motorways UK entity, which is referred to from the `sourceParty` section in the `budgetBreakdown` example above, using the `name` and `id` from the entry in the `parties` section.
```
party_1 > party

duncandewhurst commented on docs/guidance/example.md:
```diff
@@ -68,70 +99,71 @@ In this example, the overall budget for the project covers 3 years and is broken
    :jsonpointer: /projects/0/parties/0
    :expand: address, contactPoint, roles
-   :title: party
+   :title: party_1
```
party_1 > party

duncandewhurst commented on docs/guidance/example.md:
```diff
@@ -68,70 +99,71 @@ In this example, the overall budget for the project covers 3 years and is broken
+The `documents` array is used to provide information and links to documents and documentation relating to the infrastructure project. During different phases of the project, different document types are expected.
 
-In this example, an environmental impact assessment is provided. The `documentType` field is used to categorize the document against the [Document Type codelist](../../../../reference/codelists/#documenttype).
+In the document_2 example below, an environmental impact assessment is provided. The `documentType` field is used to categorize the document against the [Document Type codelist](../../../../reference/codelists/#documenttype).
```
document_2 > document

duncandewhurst commented on docs/guidance/example.md:
```diff
@@ -68,70 +99,71 @@ In this example, the overall budget for the project covers 3 years and is broken
    :jsonpointer: /projects/0/documents/1
    :expand: address, contactPoint, roles
-   :title: document
+   :title: document_2
```
document_2 > document

duncandewhurst commented on docs/guidance/example.md:
```diff
@@ -68,70 +99,71 @@ In this example, the overall budget for the project covers 3 years and is broken
 The `contractingProcesses` array is used to provide information about each contracting process associated with the project, including summary information, a list of modifications and a list of OCDS releases.
 
-This example provides information about a contracting process for the design of the motorway upgrade, with one related document (a tender notice) and two related OCDS releases, one for the tender and one for the contract award.
+In the contractingProcess_1 example given below, information is given about a contracting process for the design of the motorway upgrade, with one related document in the `documents` array (a tender notice) and two related OCDS `releases`, one for the tender and one for the contract award.
```
contractingProcess_1 > contractingProcess

given below > given

duncandewhurst commented on docs/guidance/example.md:
```diff
@@ -68,70 +99,71 @@ In this example, the overall budget for the project covers 3 years and is broken
    :jsonpointer: /projects/0/contractingProcesses/0
    :expand: summary, documents, releases
-   :title: contractingProcess
+   :title: contractingProcess_1
```
contractingProcess_1 > contractingProcess

duncandewhurst commented on docs/guidance/example.md:
```diff
@@ -68,70 +99,71 @@ In this example, the overall budget for the project covers 3 years and is broken
 Each contracting process includes a `modifications` array which is used to list any changes to the duration, price, scope or other significant aspect of the contracting process.
 
-This example shows a change in duration using the `oldContractPeriod` and `newContractPeriod` fields.
+The modification_1 example below shows a change in duration using the `oldContractPeriod` and `newContractPeriod` fields.
```
modification_1 > modification

duncandewhurst commented on docs/guidance/example.md:
```diff
@@ -68,70 +99,71 @@ In this example, the overall budget for the project covers 3 years and is broken
    :jsonpointer: /projects/0/contractingProcesses/2/summary/modifications/0
    :expand: oldContractPeriod, newContractPeriod
-   :title: modification
+   :title: modification_1
```
modification_1 > modification

duncandewhurst reviewed (changes requested):
Thanks @pindec this is really great. I took the opportunity to do a more detailed review of the language used in the worked example to align it with the rest of the documentation.

I also updated the PR description so that the associated issues will be automatically closed (see the  [Github documentation](https://help.github.com/en/github/managing-your-work-on-github/closing-issues-using-keywords) for how to do that)

Regarding #143, I think the publication policy deserves it's own section (rather than only being mentioned in the worked example).

Please could you add another page under implementation guidance named 'Guidance for data users' which can:

* explain that it is important to provide guidance for users alongside the data
* explain that this should be referenced in the `publicationPolicy` field
* link out to the OCDS guidance on publication policies and the OCDS publication policy template.

Please also update the link to the publication policy guidance from the worked example to point to this new page.

Thanks!

duncandewhurst commented:
@pindec please also add a 0.9.2 entry to the [changelog](https://standard.open-contracting.org/infrastructure/latest/en/reference/changelog/) with a bullet linking to each issue addressed by this PR and describing the change that addressed (see the [OCDS changelog](https://standard.open-contracting.org/latest/en/schema/changelog/) for examples)

Thanks!

pindec committed:
Update example.md

Amending as per @duncandewhurst's review.

pindec committed:
Fixes #143 additional with new guidance for data users page

pindec committed:
adding 0.9.2 entry to changelog https://github.com/open-contracting/infrastructure/pull/159#issuecomment-546102141

duncandewhurst commented on docs/guidance/example.md:
```diff
@@ -62,28 +62,30 @@ Each project object contains the following sections:
 
 This section provides contextual information about the infrastructure project, including:
 
+* a project identifier in the `id` field. To make the project identifier globally unique, a project identifier prefix must be added to the local identifier for the project. Project identifier prefixes are assigned by the OC4IDS helpdesk. For more information on project identifiers, refer to the [project identifiers guidance](https://github.com/open-contracting/infrastructure/pull/159/identifiers. ...
```
Please correct the URL to the project identifiers guidance.

I think Github did something clever with my proposed text and changed the relative URL to an absolute one for the PR (I should have wrapped it in backticks)

duncandewhurst commented on docs/reference/changelog.md:
```diff
@@ -1,5 +1,13 @@
+## [0.9.2] - 2019-10-25
+
+### Documentation
+* Update example file [#146](https://github.com/open-contracting/infrastructure/pull/159/commits/cf162cf2e24f3c0cd4489b9a7ff2089c0e11dd3f) - add 'publicAuthority' role and [#136](https://github.com/open-contracting/infrastructure/pull/159/commits/3b58a49139a72b56e469a8eefa2e688ce68b17fd) -   project identifier prefix.
```
Please put issue numbers at the start of the bullets and make them links to the issue, rather than the commit. You can also group similar bullets, e.g.

```
* Updates to worked example:
  * [#146](https://github.com/open-contracting/infrastructure/issues/146) - add missing 'publicAuthority' role
  * [#136](https://github.com/open-contracting/infrastructure/issues/136) - add missing project identifier prefix
```

Please update the other bullets too.

Thanks!

duncandewhurst commented on schema/project-level/project-package-schema.json:
```diff
@@ -91,7 +91,7 @@
     "publicationPolicy": {
       "title": "Publication policy",
-      "description": "A link to a document describing the publisher's [publication policy](https://standard.open-contracting.org/latest/en/implementation/publication_policy/).",
+      "description": "A link to a document describing the publisher's [guidance for data users](https://standard.open-contracting.org/infrastructure/{{version}}/{{lang}}/guidance/guidance_for_data_users.md).",
```
Sphinx doesn't do it's clever URL magic on the schema, so URLs need to be the actual URL of the built page, rather than the raw markdown file:

```suggestion
      "description": "A link to a document describing the publisher's [guidance for data users](https://standard.open-contracting.org/infrastructure/{{version}}/{{lang}}/guidance/guidance_for_data_users/).",
```

duncandewhurst reviewed (changes requested):
Thanks @pindec just a few last bits to tidy!

pindec committed:
adding changes from PR

pindec committed:
Update schema/project-level/project-package-schema.json

Fixing guidance for data users URL

Co-Authored-By: Duncan Dewhurst <duncan.dewhurst@opendataservices.coop>

duncandewhurst commented on docs/guidance/example.md:
```diff
@@ -62,7 +62,7 @@ Each project object contains the following sections:
 
-* a project identifier in the `id` field. To make the project identifier globally unique, a project identifier prefix must be added to the local identifier for the project. Project identifier prefixes are assigned by the OC4IDS helpdesk. For more information on project identifiers, refer to the [project identifiers guidance](https://github.com/open-contracting/infrastructure/pull/159/identifiers. ...
+* a project identifier in the `id` field. To make the project identifier globally unique, a project identifier prefix must be added to the local identifier for the project. Project identifier prefixes are assigned by the OC4IDS helpdesk. For more information on project identifiers, refer to the [project identifiers guidance](https://standard.open-contracting.org/infrastructure/latest/en/guidance/ ...
 
```
I assume you've used an absolute link because you couldn't get a relative link to a HTML bookmark to work.

I had the same problem in https://github.com/open-contracting/infrastructure/pull/166 so I've asked @kindly @Bjwebb whether this is possible, otherwise I think it would be better to use the relative link to the parent page, so that it works in translated versions etc.

pindec replied:
Thanks @duncandewhurst, I've now replaced that absolute link for a relative one (../../../../guidance/identifiers/#globally-unique-project-identifiers)

pindec committed:
Update example.md

jpmckinney commented on schema/project-level/project-package-schema.json:
```diff
@@ -91,7 +91,7 @@
     "publicationPolicy": {
       "title": "Publication policy",
-      "description": "A link to a document describing the publisher's [publication policy](https://standard.open-contracting.org/latest/en/implementation/publication_policy/).",
+      "description": "A link to a document describing the publisher's [guidance for data users](https://standard.open-contracting.org/infrastructure/{{version}}/{{lang}}/guidance/guidance_for_data_users/).",
```
In https://crm.open-contracting.org/issues/5167 I was leaning toward simply 'user guide' (or 'data user guide'). Is 'guidance for data users' strongly preferred?

duncandewhurst replied:
I can't remember why I thought this was better.

I'm happy with 'data user guide'.

jpmckinney commented on docs/guidance/guidance_for_data_users.md:
```diff
@@ -0,0 +1,9 @@
+# Guidance for data users
+
+Publishing OC4IDS involves making choices including what data, projects and documents are included and/or excluded, and how to classify and categorize different items.
```
What does "classify and categorize different items" refer to?

duncandewhurst replied:
This wording is taken from the publication policy guidance in the OCDS documentation.

I suggest changing this to:

> ...and how to map existing data to the fields in OC4IDS.

jpmckinney committed:
Copy-edits

Worked example
- use non-normative words on non-normative page
- fix prepositions
- each -> the, when in the context of each project
- consistent order for the functions of the project package
- consistent order for fields and their descriptions
- given -> below for clarity
- consistent punctuation
- consistent pre-example sentence
- capitalize Helpdesk (also on Support page)

Guidance for data users page
- re-use content from template

jpmckinney reviewed (commented):
I left two comments and made a commit with some copy-edits.

jpmckinney committed:
Merge 0.9-dev into documentation_fixes

duncandewhurst commented:
@pindec please could you action the two review comments from James? Thanks!

pindec committed:
Update guidance_for_data_users.md

pindec committed:
Update project-package-schema.json

jpmckinney committed:
Rename data user guide file

- Use "data user guide" more consistently throughout
- Edit changelog for clarity

jpmckinney committed:
Merge branch '0.9-dev' into documentation_fixes
